### PR TITLE
Fix changelog checks

### PR DESCRIPTION
**Pull request by mexanick**
Use bash script instead of file-existence-action, as that action doesn't seem to have an option "check if at least one exists"

Tested manually with local run of github-actions.
Currently add "no-changelog-needed" label, but can remove and add changelog for automated test.
Not sure if this propagate to an actual changelog though.

### CODE REVIEW

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -6,19 +6,30 @@ on:
...
 
-      - name: Check for news fragment
-        if: ${{ ! contains( github.event.pull_request.labels.*.name, 'no-changelog-needed')}}
-        uses: andstor/file-existence-action@v3
-        with:
-          files: ${{ env.FRAGMENT_NAME }}
-          fail: true
+      - name: Check for at least one changelog fragment
+        if: ${{ !contains(github.event.pull_request.labels.*.name, 'no-changelog-needed') }}
+        run: |
```
Comment (maxnoe) [inline .github/workflows/changelog.yml:22]: Couldn't this just be:

```
find docs/changes -type f | grep -P 'docs/changes/2411\.(feature|bugfix|api|datamodel|optimization|maintenance)\.rst'
```
?

Comment (mexanick) [reply .github/workflows/changelog.yml:22]: automatically did in pure bash, but this is indeed much neater.

**Review by maxnoe — APPROVED** [review]

**Review by LukasBeiske — APPROVED** [review]

File: .github/workflows/changelog.yml [hunk 2]
```diff
@@ -6,19 +6,19 @@ on:
...
           fetch-depth: 0
 
-      - name: Check for news fragment
-        if: ${{ ! contains( github.event.pull_request.labels.*.name, 'no-changelog-needed')}}
-        uses: andstor/file-existence-action@v3
-        with:
-          files: ${{ env.FRAGMENT_NAME }}
-          fail: true
+      - name: Check for at least one changelog fragment
+        if: ${{ !contains(github.event.pull_request.labels.*.name, 'no-changelog-needed') }}
```
Comment (Tobychev) [inline .github/workflows/changelog.yml:21]: Are you absolutely sure that no space after "!" works as you want? The evaluation rules for bash are very fickle and specific.

Comment (mexanick) [reply .github/workflows/changelog.yml:21]: yes. This PR has the "no-changelog-needed" label and passes the changelog check despite it has no changelog entry.

Comment (maxnoe) [reply .github/workflows/changelog.yml:21]: This is not bash, it is githubs expression language

**Pull request merged**